src/resources/gpio_linux.cc

```diff
@@ -148,16 +128,24 @@ static void fill_settings(gpiod_line_settings* settings, bool pull_up, bool pull
   gpiod_line_settings_set_output_value(settings, output_value);
 }
 
+Mutex* GpioPinResource::mutex_ = OS::allocate_mutex(30, "GpioPinResource");
```

Comment:
You have null checks for all the mallocs in other places, but not here. Maybe we should just get rid of the others. Malloc should not return null in this context.

Reply:
Agreed. It's a bit inconsistent.
I tried to keep them where I can easily return the "MALLOC_FAILED", but here it's just hard to do...